reaction_cime/reaction_cime_api.py

```diff
@@ -98,11 +104,32 @@ def handle_dataset_cache(id, cols=None, x_channel="x", y_channel="y"):
 
 
 # region --------------- process dataset ------------------
+def process_dataset(path: Path, save_name: str, project_id, dbo):
+    try:
+        _log.info("Processing dataset %s" % save_name)
+        start_time = time.time()
+        dbo.save_dataframe(path, save_name, 1_000, project_id)
+
+        # create default constraints file
+        save_poi_constraints(project_id, dbo=dbo)
+        save_poi_exceptions(project_id, dbo=dbo)
+
+        delta_time = time.time() - start_time
+        _log.info("--- took %i min %f s to upload file %s" % (delta_time / 60, delta_time % 60, save_name))
+
+        os.remove(path)
+    except Exception as e:
+        with create_session() as session:
+            project = session.query(Project).get(project_id)
+            if project:
+                project.file_status = f"Error {str(e)}"
+            session.commit()
```

Comment:
We probably want the os.remove in a finally block?

Reply:
I think we have it similiar in cime, the file should be deleted **after** successful processing. In case of error it should probably be deleted with the project when the user wants to? This way it is more in sync with what you see and would allow retry with different params or stuff like that

Reply:
But then you also need the corresponding logic for cleaning it up after the server restarts: https://github.com/datavisyn/dv_cime/blob/develop/dv_cime/__init__.py#L27 and https://github.com/datavisyn/dv_cime/blob/develop/dv_cime/after_server_started.py.